### PR TITLE
Keys_in_react lesson: Fix typos and grammar mistakes

### PR DESCRIPTION
**Pull request by Eveieve**
## Because

The lesson contains some typos and grammar mistakes.

## This PR

* fixes typos
* fixes grammar mistakes

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/getting_started_with_react/keys_in_react.md
Comment (xandora) [inline react/getting_started_with_react/keys_in_react.md:5]: We use "Sentence case" as per the [Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#headings).

```suggestion
### Lesson overview
```

File: react/getting_started_with_react/keys_in_react.md
Comment (xandora) [inline react/getting_started_with_react/keys_in_react.md:60]: Should we expand on the why here?

**Review by xandora — CHANGES_REQUESTED** [review]
Looks good to me, a couple things to fix up and perhaps an improvement point around that keys section. :)

File: react/getting_started_with_react/keys_in_react.md
Comment (Eveieve) [reply react/getting_started_with_react/keys_in_react.md:5]: Hi, I made sure to check previous lessons on this but I've found that all other lessons use `Lesson Overview`, even in JS courses. 
image

Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:5]: you're absolutely right and the sentence case is a recent change, hence why we try to catch these on PR's. Would you be willing to change the headers in this lesson?

Comment (Eveieve) [reply react/getting_started_with_react/keys_in_react.md:5]: I didn't know it was a recent change. Yes, I'll change all the headers in this lesson.

File: react/getting_started_with_react/keys_in_react.md
Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: I was just going through this lesson and agree it would be nice to add "why we should be careful when working with `index` as a key"

Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: what do you reckon @xandora ?

**Review by xandora — APPROVED** [review]

File: react/getting_started_with_react/keys_in_react.md
Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: @Eveieve did you see the above? 

Some notes: 
- the headers should all be fixed already with the NodeJS path being fully updated to sentence-case last week
- I think it's good to proceed with the above change; add a note that the `why` will be covered in the assignment section. 
- there's a merge conflict to look into before we can merge it :)

Comment (Eveieve) [reply react/getting_started_with_react/keys_in_react.md:60]: @ManonLef Hi, I left a comment above but I think there was a problem with this 'pending' next to my name. I think that's what's causing the merge conflict. 

image

I will post my suggestion here again to answer your second question: 

```suggestion
<span id="index-as-key">Additionally, if you're sure the list will remain unchanged throughout the application's life,  you can use the array index as a key. But this is [not recommended](https://react.dev/learn/rendering-lists#why-does-react-need-keys) since it can lead to confusing bugs if the list *does* change with any items deleted, inserted or rearranged. 
```
To your first question: I didn't change any headers for this PR, did you mean to just let me know about it?

Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: I think I mentioned it because I asked you to change them earlier somewhere in this PR. 
If it had `pending` that's probably a review you started but didn't submit :) 

When I referred to `the above` your `pending` wasn't there yet so I meant the discussion me and Xandora had about it:

Screenshot 2023-07-31 at 20 42 24

The resource is shared later in the lesson so linking to it here as well doesn't feel like the best thing to do.

Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: I like it! I'm just not sure about the last sentence:

> More on why this is will be covered in the reading later.

How do you feel about:

> You will learn more about this in the assignment section's linked articles. 

?

Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: You can commit your own actually. I will do it now and do a final suggestion. Sorry about that

File: react/getting_started_with_react/keys_in_react.md
Comment (ManonLef) [inline react/getting_started_with_react/keys_in_react.md:60]:
```suggestion
<span id="index-as-key">Additionally, if you're sure the list will remain unchanged throughout the application's life,  you can use the array index as a key. However, this is not recommended since it can lead to confusing bugs if the list changes when items are deleted, inserted, or rearranged. You will learn more about this in the assignment section's linked article.</span>
```

**Review by ManonLef — CHANGES_REQUESTED** [review]
I edited it to mention `article` since there's only one article. I also thought the sentences didn't really 'flow' nicely. 

We rarely make something bold so I removed that for I didn't think it was needed here on the word `does`. If anything it could be on `not recommended` but again, I think it's not needed. 

Would love to hear your thoughts.

File: react/getting_started_with_react/keys_in_react.md
Comment (ManonLef) [reply react/getting_started_with_react/keys_in_react.md:60]: I should have paid better attention when I gave the previous feedback. I'm sorry about coming with yet another suggestion. Thanks for your time!

**Review by ManonLef — APPROVED** [review]

**Comment by ManonLef** [comment]
This is ready to merge once the merge conflict is addressed :)

**Review by ManonLef — APPROVED** [review]

**Comment by ManonLef** [comment]
Thank you @Eveieve 🎉

**Pull request merged**
